core/rio/turtle/src/test/java/org/eclipse/rdf4j/rio/turtle/TurtleWriterTest.java

```diff
@@ -672,6 +674,20 @@ public void testBlankNodeInlining_indirectCircularReferenceWithIRI() throws Exce
 		assertTrue(Models.isomorphic(expected, actual));
 	}
 
+	@Test
+	public void testIriNamespace() throws Exception {
+		Model model = new DynamicModelFactory().createEmptyModel();
+		String prefix = "foo-bar";
+		String ns = "foo:this.is.my.bar.";
+		model.setNamespace(prefix, ns);
+		model.add(vf.createIRI(ns, "lala"), vf.createIRI(ns, "lulu"), vf.createIRI(ns, "lolo"));
```

Comment:
So the full IRI of the subject would be `foo:this.is.my.bar.lala`. I'm still not convinced that this is a valid IRI, but I'll try to take a closer look at the spec again.

Reply:
Had a look now and found the example section of the URI spec. Since all URIs are valid IRIs then this should hold.

https://datatracker.ietf.org/doc/html/rfc3986#section-1.1.2

This example here is similar to your test case: `news:comp.infosystems.www.servers.unix`